autoload/highlightedyank.vim

```diff
@@ -85,6 +85,55 @@ function! s:yank_normal(count, register) abort "{{{
   endif
 endfunction
 "}}}
+function! highlightedyank#autocmd_highlight() abort "{{{
+  let view = winsaveview()
+  let region = {}
+  let region.head = getpos("'[")
+  let region.tail = getpos("']")
+  let char = getline(region.tail[1])[region.tail[2] - 1]
+  let linecontent = v:event.regcontents[len(v:event.regcontents) - 1]
+  let tailchar = linecontent[len(linecontent) - 1]
+  if char != tailchar
+    let region.tail[2] = region.tail[2] - 1
+  endif
+  if v:event.regtype == 'v'
+    let motionwise = 'char'
+  elseif v:event.regtype == 'V'
+    let motionwise = 'line'
+  elseif v:event.regtype =~ "\<c-v>"
+    let motionwise = 'block'
+  else
+    let motionwise = ''
+  endif
```

Comment:
I do not have much experience about Neovim. So I'm not so sure about v:event, but if it should be `v`, `V`, or `<C-v>` then it is available as it is.
```vim
let motionwise = v:event.regtype
```

Both `{char|line|block}` and `{v|V|\<C-v>}` are acceptable as motionwise :)